### ISSUE TITLE
"/system-update not found" after reboot

### ISSUE THREAD

**Issue by almereyda** [rpm-software-management/dnf-plugin-system-upgrade#1]
When issuing

```
dnf fedup --releasever=22 download
dnf fedup --releasever=22 reboot
```

the system boots up, but reboots immediately again, because `journalctl -b -1` tells me clearly what doesn't work

```
Jun 28 00:11:55 neocortex systemd[1]: Starting System Upgrade...
Jun 28 00:11:55 neocortex systemd[1]: Started System Upgrade.
Jun 28 00:11:55 neocortex systemd[1]: Starting firewalld - dynamic firewall daemon...
Jun 28 00:11:55 neocortex systemd[1]: Starting Updates the operating system whilst offline...
Jun 28 00:11:55 neocortex systemd[1]: Started Updates the operating system whilst offline.
Jun 28 00:11:55 neocortex systemd[1]: Started Drop Read-Ahead Data.
Jun 28 00:11:55 neocortex systemd[1]: Starting System Update.
Jun 28 00:11:55 neocortex systemd[1]: Reached target System Update.
Jun 28 00:11:55 neocortex systemd[1]: Starting Stop Read-Ahead Data Collection 10s After Completed Startup.
Jun 28 00:11:55 neocortex systemd[1]: Started Stop Read-Ahead Data Collection 10s After Completed Startup.
Jun 28 00:11:55 neocortex pk-offline-update[651]: failed to read /var/lib/PackageKit/prepared-update: No offline updates have been prepared
Jun 28 00:11:55 neocortex kernel: nf_conntrack version 0.5.0 (65536 buckets, 262144 max)
Jun 28 00:11:55 neocortex kernel: ip6_tables: (C) 2000-2006 Netfilter Core Team
Jun 28 00:11:55 neocortex kernel: Ebtables v2.0 registered
Jun 28 00:11:55 neocortex kernel: bridge: automatic filtering via arp/ip/ip6tables has been deprecated. Update your scripts to load br_netfilter if you need this.
Jun 28 00:11:55 neocortex systemd[1]: Starting D-Bus System Message Bus...
Jun 28 00:11:55 neocortex systemd[1]: Started D-Bus System Message Bus.
Jun 28 00:11:55 neocortex systemd[1]: Started firewalld - dynamic firewall daemon.
Jun 28 00:11:55 neocortex systemd[1]: Startup finished in 754ms (kernel) + 1.906s (initrd) + 2.292s (userspace) = 18.167s.
Jun 28 00:11:56 neocortex dnf[648]: [Errno 2] Datei oder Verzeichnis nicht gefunden: '/system-update'
Jun 28 00:11:56 neocortex systemd[1]: fedup-system-upgrade.service: main process exited, code=exited, status=1/FAILURE
Jun 28 00:11:56 neocortex systemd[1]: Unit fedup-system-upgrade.service entered failed state.
Jun 28 00:11:56 neocortex systemd[1]: Triggering OnFailure= dependencies of fedup-system-upgrade.service.
Jun 28 00:11:56 neocortex systemd[1]: fedup-system-upgrade.service failed.
Jun 28 00:11:56 neocortex systemd[1]: Starting Show Plymouth Reboot Screen...
```

[I'm currently running Fedora 21 with systemd 216](https://github.com/rhinstaller/fedup/issues/62).
The strange thing is
1. I had to specify the `--releasever` flag for the reboot command, too, else my current one would end up in `.fedup-system-upgrade`
2. Trying to get around this, I created the symlink by hand, but then cannot run `dnf fedup reboot` anymore due to it existing already.

But why does it then reboot if initially non-existing, therefore create, but doesn't find it in the offline environment? Let me try to prepare the env by hand and reboot manually ...

**Comment by almereyda** [comment]
... which didn't work, either. @wgwoods Any ideas?
Is maybe pk-offline-update deleting the symlink before the dnf plugin can take over? How can I debug when it get's lost?
Appearantly it's existing at reboot so the `systemd-system-update-generator` is invoked.

**Comment by almereyda** [comment]
Okay, it was the PackageKit.
I've ran `mv /usr/lib/systemd/system/packagekit-offline-update.service /usr/lib/systemd/system/packagekit-offline-update.service.off` and the installation proceeded.
The system now shows Fedora 22 with systemd 219.
During installation display degraded continuously, but it ran fine until the end.

Let's also figure out how we can automise this for future users, especially in my case WiFi seems to be broken (`wpa_supplicant` is now unused, right? I'll have to find a place for the issue, as `iwlist scan` shows my wifis, but NetworkManager doesn't.), `terminal` has a transparent window decoration and `gnome-shell` is still 3.14.

So it seems the upgrade path is working, but the proof of concept can be refined. Thanks for your work.

**Comment by almereyda** [comment]
After an external USB WiFi also couldn't see anything, despite `iwlist wlp0s20u1 scan` returned results, external LAN + `dnf update` finally turned everything consistend again.
But where did the WiFi go?
And how can these experiences help improving `dnf-plugin-fedup`?

Anyway, thanks for the upgrade path alternative; I very much appreciate being on F22 + G3.16 now.

**Comment by wgwoods** [comment]
> I had to specify the --releasever flag for the reboot command, too, else my current one would end up in .fedup-system-upgrade

Oh, whoops. Should be fixed by 7f83963.

> Is maybe pk-offline-update deleting the symlink before the dnf plugin can take over?

Correct - should be fixed by 2cd7fa8. Thanks!

> in my case WiFi seems to be broken ... `dnf update` finally turned everything consistend again.

Strange - did DNF not download/install packages from the updates repos during your system upgrade?

**Comment by almereyda** [comment]
In the name of the other, future users, thanks for your updating commits.

> Strange - did DNF not download/install packages from the updates repos during your system upgrade?

I've predownloaded and ran an offine update.
Then it must have rebooted during installation of packages and left the system in inconsistent state. Unfortunately I wasn't able to see when, due to the screen distortions which happend during the process.

Also a recent `iwl7260-firmware` update didn't bring WiFi back, despite `iwlist scan` shows the networks around. Could you point me in direction of the Networking / NetworkManager / GNOME NetworkManager UI people of Fedora to help me isolate the issue?

**Comment by wgwoods** [comment]
Not sure, but I'd try https://ask.fedoraproject.org/ for kernel/wireless questions. Good luck!

**Comment by almereyda** [comment]
> Good luck!

Thanks, [will try](https://ask.fedoraproject.org/en/question/72230/intel-wifi-stopped-working-after-applying-dnf-plugin-fedup/).

Just for sake of documentation, because of the interrupted offline setup and the weird reboot, now I've found out I also had to run `[root@neocortex yum.repos.d]# mv fedora.repo.rpmnew fedora.repo` to get all the 525 remaining updates. Let's see if it fixes WiFi.
